### PR TITLE
refactor: replace used options impl with options_cli

### PR DESCRIPTION
**Pull request by jackgerrits**
- Since opts are provided by the core lib now spanning tree was removed from the solution

### CODE REVIEW

File: cluster/spanning_tree_main.cc [hunk 1]
```diff
@@ -20,51 +24,48 @@ int getpid() { return (int)::GetCurrentProcessId(); }
...
+
+  VW::config::options_cli opts(std::vector<std::string>(argv + 1, argv + argc));
+
+  VW::config::option_group_definition desc("Spanning Tree");
+  desc.add(VW::config::make_option("nondaemon", nondaemon).help("Run spanning tree in foreground"));
+  desc.add(VW::config::make_option("help", help).short_name('h').help("Print help message"));
+  desc.add(VW::config::make_option("port", port)
+               .short_name('p')
+               .default_value(26543)
+               .help("Port number for spanning tree to listen on"));
```
Comment (bassmang) [inline cluster/spanning_tree_main.cc:48]: nit: can we keep make this `.add()...add()...add()`

File: test/unit_test/options_boost_po_test.cc [hunk 2]
```diff
@@ -238,20 +237,6 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(incorrect_option_type_str_for_int, T, option_types
   BOOST_CHECK_THROW(options->add_and_parse(arg_group), VW::vw_argument_invalid_value_exception);
 }
 
-BOOST_AUTO_TEST_CASE(multiple_locations_one_option)
```
Comment (bassmang) [inline test/unit_test/options_boost_po_test.cc:241]: why remove this test?

Comment (jackgerrits) [reply test/unit_test/options_boost_po_test.cc:241]: The new implementations doesn't really care about this situation, whereas boost in the past threw in this case im pretty sure. It's purely a developer side thing

**Review by bassmang — APPROVED** [review]

File: test/unit_test/options_boost_po_test.cc (hunk 2 above)
Comment (olgavrou) [reply test/unit_test/options_boost_po_test.cc:241]: is there no value in still keeping the test and not checking for throw, but just checking that it works fine?

File: vowpalwabbit/parse_args.cc [hunk 3]
```diff
@@ -1802,7 +1803,8 @@ VW::workspace* initialize_with_builder(int argc, char* argv[], io_buf* model, bo
     trace_message_t trace_listener, void* trace_context, std::unique_ptr<VW::setup_base_i> learner_builder)
 {
   std::unique_ptr<options_i, options_deleter_type> options(
-      new config::options_boost_po(argc, argv), [](VW::config::options_i* ptr) { delete ptr; });
+      new config::options_cli(std::vector<std::string>(argv + 1, argv + argc)),
```
Comment (olgavrou) [inline vowpalwabbit/parse_args.cc:1806]: can we use vw's make unique here?

Comment (jackgerrits) [reply vowpalwabbit/parse_args.cc:1806]: You can't use a deleter with `std::make_unique` or `VW::make_unique` unfortunately

**Pull request merged**